app/main/views/suppliers.py

```diff
@@ -222,13 +226,15 @@ def edit_supplier_registration_number():
                 if form.has_companies_house_number.data == "Yes":
                     data_api_client.update_supplier(
                         supplier_id=current_user.supplier_id,
-                        supplier={"companiesHouseNumber": form.companies_house_number.data.upper()},
+                        supplier={"companiesHouseNumber": form.companies_house_number.data.upper(),
+                                  "otherCompanyRegistrationNumber": None},
```

Comment:
OK, this does break it. Using `None` here kicks and API error.
`dmapiclient.errors.HTTPError: JSON was not a valid format. None is not of type 'string' (status: 400)`

Think you need to give it empty string.

Reply:
This is what the corresponding api pr is for - to allow setting back to none/null. You'll need to check that out as well to test.

Reply:
aha!

Reply:
Yep, totally works with the api branch